### PR TITLE
Docstring handling upgrade

### PR DESCRIPTION
**Pull request by mih**
Prompoted by https://github.com/datalad/datalad/issues/6247#issuecomment-979218397

- Start of a design document on docstring handling and formatting (it does not document all possibilities on purpose, because some seem arcane and redundant)
- Add support for sphinx versioning directives
- Make `REFLOW` work with indented paragraphs
- Support a simpler specification for API argument alternatives
- Use versioning directives for the 0.16 set of changes to `create_sibling_g*``

Demos:

The is how the following sources look across all kinds of rendering scenarios:

```rst
    .. versionchanged:: 0.16
       || REFLOW >>
       The API has been aligned with the some
       ``create_sibling_...||create-sibling-...`` commands of other GitHub-like
       services, such as GOGS, GIN, GitTea.<< REFLOW ||

    .. deprecated:: 0.16
       The ``dryrun||--dryrun`` option will be removed in a future release, use
       the renamed ``dry_run||--dry-run`` option instead.
       The ``github_login||--github-login`` option will be removed in a future
       release, use the ``credential||--credential`` option instead.
       The ``github_organization||--github-organization`` option will be
       removed in a future release, prefix the reposity name with ``<org>/``
       instead.
```

image

image

image

image

image

### CODE REVIEW

File: docs/source/design/docstrings.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+Docstrings
+**********
+
+.. topic:: Specification scope and status
+
+   This specification provides a partial overview of the current
+   implementation.
+
+Docstrings in DataLad source code are used and consumed in many ways. Besides
+serving as documentation directly in the sources, the are also transformed
```
Comment (adswa) [inline docs/source/design/docstrings.rst:16]:
```suggestion
serving as documentation directly in the sources, they are also transformed
```

File: docs/source/design/docstrings.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+rendered in various ways.
```
Comment (adswa) [inline docs/source/design/docstrings.rst:17]:
```suggestion
and rendered in various ways.
```

forgotten and?

**Review by adswa — APPROVED** [review]
I like this a lot! Thank you!

**Review by bpoldrack — COMMENTED** [review]
Nice. I like it.

I wonder whether the example declarations should be part of the design doc for docstrings. While technically not part of the docstring, I think it's the natural place to look up "How do properly document?"

**Comment by mih** [comment]
> Nice. I like it.
> 
> I wonder whether the example declarations should be part of the design doc for docstrings. While technically not part of the docstring, I think it's the natural place to look up "How do I properly document?"

Yes, please add them.

**Comment by mih** [comment]
Given the outburst of positivity, I will merge as soon as the tests pass. Thx for the reviews!

**Comment by mih** [comment]
Benchmarks for producing the help got slower, but it is a better help also ;-)

No impact on command startup time, and no impact on `api` import.

**Pull request merged**